src/lib/event_mapper/impl/general_aws_event_mapper.py

```diff
@@ -0,0 +1,146 @@
+
+        Args:
+            event (dict): Event object
+        """
+        pass
+
+    @abstractmethod
+    def get_message_body(self, event: dict):
+        """Returns composed message body for the given AWS event
+
+        Args:
+            event (dict): Event object
+        """
+        pass
+
+    def __get_delivery_options(self, resource_name: str) -> list[dict]:
+        """Returns delivery options section for the given resource name based on settings config.
+
+        Args:
+            resource_name (str): Name of the AWS resourcce
+
+        Returns:
+            list[dict]: List of delivery options including recipient and delivery method
+        """
+        delivery_options = []
+        recipients = {}
+        monitoring_groups = self.settings.get_monitoring_groups([resource_name])
+        recipients_settings = self.settings.get_recipients(
+            monitoring_groups, NotificationType.ALERT
+        )
+
+        for recipient_setting in recipients_settings:
+            delivery_method = recipient_setting["delivery_method"]
+            recipient = recipient_setting["recipient"]
+
+            if delivery_method not in recipients:
+                recipients[delivery_method] = []
+
+            recipients[delivery_method].append(recipient)
+
+        for delivery_method, recipients_array in recipients.items():
+            delivery_option = {
+                "delivery_method": delivery_method,
+                "recipients": recipients_array,
+                "sender_email": self.settings.get_sender_email(delivery_method),
+            }
+            delivery_options.append(delivery_option)
+
+        return delivery_options
+
+    def __to_notification_messages(
+        self, delivery_options: list[dict], message: dict
+    ) -> list[dict]:
+        """Maps delivery options and given message to the notification messages array
+
+        Args:
+            delivery_options (list[dict]): List of delivery options
+            message (dict): Message to be sent
+
+        Returns:
+            list[dict]: List of notification messages
+        """
+        notification_messages = []
+
+        for delivery_option in delivery_options:
+            notification_message = {
+                "delivery_options": delivery_option,
+                "message": message,
+            }
+            notification_messages.append(notification_message)
+
+        return notification_messages
+
+    def __get_message_subject(self, event: dict) -> str:
+        """Return message subject based on the event
+
+        Args:
+            event (dict): Event object
+
+        Returns:
+            str: Message subject
+        """
+        monitored_env_name = self.settings.get_monitored_environment_name(
+            event["account"], event["region"]
+        )
+        event_message = event["detail-type"]
+        return f"{monitored_env_name}: {event_message}"
```

Comment:
It's not critical to changed it in this PR (maybe, as a follow-up task), but I believe we need subject line to be a bit informative.
Instead of "Env1: Glue Job State Change" it's better to have
"Env1: Glue Job my-lovely-job has FAILED" (SUCCEEDED).

For users it'd be much easier to set up e-mail filters etc.

Reply:
Agree. Created a card in Trello.